Assets/Scripts/Controllers/PlayerController.cs

```diff
@@ -105,11 +108,25 @@ public void OnTriggerExit(Collider collision)
             {
                 Debug.Log("Player has left the interaction zone of an interactable object."); // TODO : Remove
                 interactable.StopInteract();
-                interactable = null;
+                 interactable = null;
             }
         }
     }
     
+    protected override void Die()
```

Comment:
Instead of duplicating the die code here,you can reuse the parent method from CharController like this, the logic is the same: